sdk/python/symbolchain/TypeParserBuilder.py

```diff
@@ -0,0 +1,98 @@
+import functools
+import importlib
+import operator
+
+from .TransactionDescriptorProcessor import TransactionDescriptorProcessor
+
+
+def _name_to_enum_value(mapping, enum_type, enum_value_name):
+	if enum_value_name not in mapping:
+		raise RuntimeError(f'unknown value {enum_value_name} for type {enum_type}')
+	return mapping[enum_value_name]
+
+
+def build_type_hints_map(transaction):
+	type_hints = {}
+	for key, hint in transaction.TYPE_HINTS.items():
+		rule_name = None
+		if hint.startswith('array['):
+			rule_name = hint
+		elif hint.startswith('enum:'):
+			rule_name = hint[len('enum:'):]
+		elif hint.startswith('pod:'):
+			rule_name = hint[len('pod:'):]
+		elif hint.startswith('struct:'):
+			rule_name = hint
+
+		if rule_name:
+			type_hints[key] = rule_name
+
+	return type_hints
+
+
+class TypeParserBuilder:
```

Comment:
not sure how to test this one, given it  uses `get_catbuffer_class` inside created parsers...
(also as above, not sure if should test this one directly)

Reply:
at minimum, you should be able to refactor tests - in model of BasicAddressTests - that are shared by both NEM and Symbol. as long as you fully cover this (base class) in those, additional tests just for it probably is not required.

Reply:
I made both TransactionTests similar to each other, but I think extracting common things will result in quite a mess